Compatibility with MySQL 8

**Describe the bug**
Building with Mysql 8 library is not currently possible.

**To Reproduce**
Steps to reproduce the behavior:
1. install Mysql8 via 'brew install mysql' 
2. build hercules as you would normally

**Expected behavior**
Build without emitting errors.

**System specs (please complete the following information):**
 - OS: macOS 10.14
 - Hercules Version 51483b4569a6018f79ae8ece69273e3611e98a89
 - Mode: either
 - Packet version: any
 - Client type: any

**Plugins used or source modifications**
None

**Additional context**
Configured with Options:
```
./configure --with-MYSQL_LIBS=/usr/local/opt/mysql/lib/libmysqlclient.dylib
```
Configure Output
```
checking for mysql_config... /usr/local/bin/mysql_config
checking for mysql_init... yes
checking mysql.h usability... yes
checking mysql.h presence... yes
checking for mysql.h... yes
checking MySQL library (required)... yes (8.0.13)
```
Make Output
```
	CC	sql.c
sql.c:95:3: error: use of undeclared identifier 'my_bool'
                my_bool reconnect = 1;
                ^
sql.c:96:54: error: use of undeclared identifier 'reconnect'
                mysql_options(&self->handle, MYSQL_OPT_RECONNECT, &reconnect);
                                                                   ^
sql.c:527:19: error: use of undeclared identifier 'my_bool'; did you mean '_Bool'?
        bind->is_null = (my_bool*)out_is_null;
                         ^~~~~~~
                         _Bool
sql.c:527:19: error: expected expression
sql.c:832:3: error: use of undeclared identifier 'my_bool'
                my_bool truncated;
                ^
sql.c:843:21: error: use of undeclared identifier 'truncated'
                        column->error = &truncated;
                                         ^
sql.c:846:8: error: use of undeclared identifier 'truncated'
                        if (truncated) {
                            ^
7 errors generated.
make[1]: *** [obj_sql/sql.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [common_sql] Error 2
```

--- a/configure
+++ b/configure
@@ -9452,6 +9452,46 @@ else
 fi
 
 
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether my_bool is supported (MySQL)" >&5
+$as_echo_n "checking whether my_bool is supported (MySQL)... " >&6; }
+if test "$cross_compiling" = yes; then :
+  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
+$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
+as_fn_error $? "cannot run test program while cross compiling
+See \`config.log' for more details" "$LINENO" 5; }
+else
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+		#ifdef WIN32
+		#include "common/winapi.h" // Needed before mysql.h
+		#endif
+		#include <mysql.h>
+		int main(int argc, char **argv){
+			my_bool val = (my_bool)0;
+			return (int)val;
+		}
+
+_ACEOF
+if ac_fn_c_try_run "$LINENO"; then :
+
+		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+
+else
+
+		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no (converting my_bool to bool)" >&5
+$as_echo "no (converting my_bool to bool)" >&6; }
+		MYSQL_CFLAGS="$MYSQL_CFLAGS -Dmy_bool=bool"
+
+
+fi
+rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
+  conftest.$ac_objext conftest.beam conftest.$ac_ext
+fi
+
+
 CPPFLAGS="$MYSQL_OLD_CPPFLAGS"
 LIBS="$MYSQL_OLD_LIBS"
 

--- a/configure.ac
+++ b/configure.ac
@@ -1597,6 +1597,28 @@ MYSQL_OLD_LIBS="$LIBS" ; LIBS="$LIBS $MYSQL_LIBS"
 MYSQL_OLD_CPPFLAGS="$CPPFLAGS" ; CPPFLAGS="$CPPFLAGS $MYSQL_CFLAGS"
 AC_CHECK_FUNC([mysql_init], [HAVE_MYSQL="yes"])
 AC_CHECK_HEADER([mysql.h], [], [HAVE_MYSQL=""])
+
+AC_MSG_CHECKING([whether my_bool is supported (MySQL)])
+AC_RUN_IFELSE(
+	[AC_LANG_SOURCE([
+		#ifdef WIN32
+		#include "common/winapi.h" // Needed before mysql.h
+		#endif
+		#include <mysql.h>
+		int main(int argc, char **argv){
+			my_bool val = (my_bool)0;
+			return (int)val;
+		}
+	])],
+	[
+		AC_MSG_RESULT([yes])
+	],
+	[
+		AC_MSG_RESULT([no (converting my_bool to bool)])
+		MYSQL_CFLAGS="$MYSQL_CFLAGS -Dmy_bool=bool"
+	]
+)
+
 CPPFLAGS="$MYSQL_OLD_CPPFLAGS"
 LIBS="$MYSQL_OLD_LIBS"
 

